### PR TITLE
solve https://github.com/nitely/nim-regex/issues/123

### PR DESCRIPTION
**Pull request by Ethosa**
So, I just remove `distinct` keyword. Also I created procedures `toRegex(Regex2)` and `toRegex2(Regex)` for convert.

### CODE REVIEW

**Comment by Ethosa** [comment]
> this makes copies, though

Maybe Nim allows to destruct `object`? 🤔

**Comment by Ethosa** [comment]
cast[Regex](regex2) is also forbidden by NIm VM
image

**Comment by Ethosa** [comment]
@nitely please see changes

File: src/regex.exe
Comment (SolitudeSF) [inline src/regex.exe:1]: ah hell naw

Comment (Ethosa) [reply src/regex.exe:1]: oh, sorry, my mistake)))

**Comment by Ethosa** [comment]
@nitely please check again

**Comment by nitely** [comment]
I think the point of using converters is you don't need to call them. There is an implicit call, like [here](https://nim-lang.org/docs/manual.html#converters). If you are calling them, then just use a func.

FIY, one way to avoid copies when there are two compatible types like this, is to define something like `type RegexT = Regex | Regex2` and use it internally, so you can pass any of those two types. But I see you are only copying on JS, which I don't mind because JS is slow anyway, and if someone cares about performance in JS they should use the native JS regex.

Also, this is really Nim regression, and it would be better if the compiler gets fixed/backported. But if tests pass, I'll merge it.

**Pull request merged**